### PR TITLE
Remove SwiftyJSON Dependency

### PR DESCRIPTION
**Pull request by l-szyszkowski**
What's done in this PR:
- Remove completely SwiftyJSON library. This library is 4yrs old and won't compile with Xcode 14.3
- Introduce JSONUtility to manage everything what was managed by SwiftyJSON
- Introduce codable structs like `CryptoData` and `TestAppSetup`

Not every json structure has reflection in dedicated codable structs because I want to keep code as similar as possible in test classes. On the other hand, some of them are too complex to manage as codable, like `messages-encoding.json` which has various types of `expectedValue`.

### CODE REVIEW

**Comment by lawrence-forooghian** [comment]
I haven't reviewed this one yet but I noticed that there are still references to SwiftyJSON in the codebase (Xcode project file and Carthage files). Could you remove those please?

**Comment by l-szyszkowski** [comment]
> I haven't reviewed this one yet but I noticed that there are still references to SwiftyJSON in the codebase (Xcode project file and Carthage files). Could you remove those please?

changed here 575505502d23a8a812b6ecfede6803223ea7e91f

**Comment by lawrence-forooghian** [comment]
@lukasz-szyszkowski does this resolve https://github.com/ably/ably-cocoa/issues/1668? If so, please can you mark it as doing so, so that when this is merged that issue gets closed?

File: Test/Test Utilities/JSON.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+extension Encodable {
+    func rawData() throws -> Data {
+        try JSONUtility().encode(self)
+    }
+}
+
+extension Data {
+    init(path: String) throws {
```
Comment (lawrence-forooghian) [inline Test/Test Utilities/JSON.swift:77]: Do we really need an extension method on `Data` for this? We're using it in one place in the codebase, and it doesn't seem like it's adding much value.

File: Test/Test Utilities/JSON.swift [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        guard let data = try? encode(model) else {
+            return nil
+        }
+        
+        return try? JSONSerialization.jsonObject(with: data) as? [String: Any]
+    }
+}
+
+extension Encodable {
+    func rawData() throws -> Data {
```
[the remark this answers is not in the document]
Comment (l-szyszkowski) [reply Test/Test Utilities/JSON.swift]: a812568931edf150a1a934944ae00aa2be86d93f

File: Test/Test Utilities/JSON.swift [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+//  JSON.swift
+//  Ably
+//
+//  Created by Łukasz Szyszkowski on 14/04/2023.
+//  Copyright © 2023 Ably. All rights reserved.
+//
+
+import Foundation
+
+final class JSONUtility {
```
[the remark this answers is not in the document]
Comment (l-szyszkowski) [reply Test/Test Utilities/JSON.swift]: c37246047684c8aad5f4ccb5ebe7c971da395bd3

File: Test/Test Utilities/JSON.swift (hunk 1 above)
Comment (l-szyszkowski) [reply Test/Test Utilities/JSON.swift]: 9042e2be51f449d9628819c77b0e10f0a8410986

**Review by lawrence-forooghian — APPROVED** [review]

**Pull request merged**